optuna/study.py

```diff
@@ -259,8 +259,12 @@ def optimize(
             gc_after_trial=True  # type: bool
     ):
         # type: (...) -> None
-        """Optimize an objective function.
-
+        """ Optimize an objective function using Bayesian Optimization which builds a surrogate 
+            model for the objective and quantifies the uncertainty in that surrogate using 
+            TPE (Tree-structured Parzen Estimator), and then uses an acquisition function defined 
+            from this surrogate to decide where to sample.
+            See also :class:`~optuna.samplers.TPESampler` for more details on 'TPE'.
```

Comment:
Thanks @arpitkh101 for the PR. The intention to clarify some of the logic behind the optimization process seem right, but we shouldn't really explain the TPE algorithm in the documentation of this method (since samplers are configurable). Let me suggest some changes. How about explaining that the `Study` holds a sampler, which defines the optimization algorithm, and that the sampler can be specified in `optuna.create_study` ? We could then leave a reference to `optuna.create_study`, maybe also mention that the default sampler is TPE and add a reference to the TPE sampler as well.

Reply:
Thanks @hvy for your review. I absolutely agree with with your point that the samplers are configurable and TPE shouldn't be there . I have described the basic idea about Bayesian optimization  and what if I just use "Sampler" instead of "TPE (Tree-structured Parzen Estimator)" and then we could mention that the sampler can be specified in optuna.create_study and TPE is a default choice, in the next line. Would it be right?

Reply:
> I have described the basic idea about Bayesian optimization and what if I just use "Sampler" instead of "TPE (Tree-structured Parzen Estimator)"

Actually, we have samplers that are not based on BO, e.g. the [random sampler](https://github.com/optuna/optuna/blob/master/optuna/samplers/random.py) and the [grid search sampler coming up](https://github.com/optuna/optuna/pull/665). So, I believe it's less confusing if we avoid mentioning BO here. What do you think?

Reply:
Yes, you are right since the other samplers are not based on BO it shouldn't be mentioned here. 
So here are some changes to the documentation. Please review them so I could commit these changes. 
"It optimizes the objective function by choosing a suitable set of Hyperparameter values from a given range. `Study` also includes a Sampler which implements the task of value suggestion based on a specified distribution. The Sampler could be defined `optuna.create_study` and the default choice for the sampler is `TPE` " then i could add the reference.

Reply:
Thanks for the suggestion. It basically LGTM. Some suggestion though.

"~It optimizes~Optimize the objective function by choosing a suitable set of ~H~hyperparameter values from a given range. ~Study also includes~Uses a ~S~sampler which implements the task of value suggestion based on a specified distribution. The ~S~sampler ~could be defined~ is specified in ~optuna.create_study~:func:`~optuna.study.create_study` and the default choice for the sampler is TPE "